tests/app/questionnaire/test_placeholder_transforms.py

```diff
@@ -7,36 +7,57 @@
 
 
 @pytest.mark.parametrize(
-    "number, currency, expected",
+    "value, currency, decimal_limit, formatted_currency",
     (
-        ("11", "GBP", "£11.00"),
-        ("11.99", "GBP", "£11.99"),
-        ("11000", "USD", "US$11,000.00"),
-        (0, None, "£0.00"),
-        (0.00, None, "£0.00"),
+        (2, "GBP", 6, "£2.00"),
+        ("11", "GBP", 2, "£11.00"),
+        ("11.99", "GBP", 2, "£11.99"),
+        ("11000", "USD", 2, "US$11,000.00"),
+        (0, None, 2, "£0.00"),
+        (0.00, None, 2, "£0.00"),
+        (11000, "USD", 2, "US$11,000.00"),
+        (Decimal("123.45"), "GBP", 1, "£123.4"),
+        (Decimal("2.14564"), "GBP", 6, "£2.14564"),
+        (Decimal("2.1"), "GBP", 1, "£2.1"),
+        (Decimal("2.1"), "GBP", None, "£2.10"),
+        (Decimal("3000"), "JPY", 0, "JP¥3,000"),
+        (Decimal("1.1"), "GBP", 6, "£1.10"),
+        (Decimal("3000.445"), "GBP", 6, "£3,000.445"),
+        (Decimal("3000.44545"), "GBP", None, "£3,000.44545"),
+        ("", "", "", ""),
     ),
 )
-def test_format_currency(number, currency, expected, transformer):
-    assert transformer().format_currency(number, currency or "GBP") == expected
+def test_format_currency(
```

Comment:
Grouping the tests by decimal length and currency would help.

Reply:
Grouping now 👍